src/m3ninx/index/segment/builder/bytes_slice_iter.go

```diff
@@ -22,31 +22,36 @@ package builder
 
 import (
 	"bytes"
-	"sort"
 
 	"github.com/m3db/m3/src/m3ninx/index/segment"
+	"github.com/m3db/m3/src/m3ninx/postings"
 
 	"github.com/twotwotwo/sorts"
 )
 
+type uniqueField struct {
+	field        []byte
+	postingsList postings.List
+}
+
 // OrderedBytesSliceIter is a new ordered bytes slice iterator.
 type OrderedBytesSliceIter struct {
 	err  error
 	done bool
 
 	currentIdx    int
-	current       []byte
-	backingSlices *sortableSliceOfSliceOfByteSlicesAsc
+	current       uniqueField
+	backingSlices *sortableSliceOfSliceOfUniqueFieldsAsc
 }
 
-var _ segment.FieldsIterator = &OrderedBytesSliceIter{}
+var _ segment.FieldsPostingsListIterator = &OrderedBytesSliceIter{}
```

Comment:
nit: Since `OrderedBytesSliceIter` doesn't do generic byte slice ordering any more, maybe better to unexport `OrderedBytesSliceIter` and call it something like `newFieldsIterator(...)` for constructor and `fieldsIterator` for the struct?

Reply:
Renamed to `orderedFieldsPostingsListIter` and `newOrderedFieldsPostingsListIter`.